lib/ansible/plugins/action/__init__.py

```diff
@@ -1104,7 +1104,7 @@ def _get_diff_data(self, destination, source, task_vars, source_file=True):
 
         if not peek_result.get('failed', False) or peek_result.get('rc', 0) == 0:
 
-            if peek_result.get('state') == 'absent':
+            if peek_result.get('state') is None:
```

Comment:
i would do both `in ('absent', None)` as there are cases in which 'absent' would be a correct return.

Reply:
I don't think there are. `peek_result` is a result of the file module being executed with `_diff_peek=True` which just checks if the file is binary and calls `exit_json`. From there `add_path_info` is only method that modifies `state` in the result and no longer sets `absent`.

Having said that doing `in ('absent', None)` might be more defensive, so I'd be +1 for that.

Reply:
you also have to take into account actions that have 'remote_src=true' as they might not rely on peek